---
title: "Relatorio Parcial"
---


```{r}
#BiocManager::install("IlluminaHumanMethylationEPICanno.ilm10b4.hg19")
#BiocManager::install("missMethyl")
#BiocManager::install("minfiData")

```

```{r,message=FALSE,warning=FALSE}
library(limma)
library(minfi)
library(IlluminaHumanMethylationEPICmanifest)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(RColorBrewer)
library(readr)
library(maxprobes)
#tidyverse::tidyverse_packages()
#install.packages("remotes")
#remotes::install_github("markgene/maxprobes")
```

# Carregando os dados

```{r}
load("../../data/rgSetSample.rds")
```

```{r}
dim(rgSet)
```

# Passo a passo do Cross Package

## Banco de dados amostrado

```{r}
annEPIC <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
#dim(annEPIC)
```
46 informações de 865859 ilhas.

```{r}
sampleNames(rgSet)
```
Cada uma dessas amostras apresenta os arquivos green e red

```{r}
rgSet
```

```{r}
qcReport(rgSet, pdf = "qcReport.pdf")
```

## Controle de qualidade

```{r}
detP <- data.frame(detectionP(rgSet))
```

```{r}
write.csv(detP, "../../data/pvalor.csv", row.names=T)
```

```{r}
detP = read.csv(file = "../../data/pvalor.csv", row.names = 1)
```

```{r}
barplot(colMeans(detP), las=2, cex.names=0.5, ylim=c(0,0.0012))
abline(h=0.001,col="red")
```
Legenda do gráfico: Média dos p-valores detectados nas CpG's para cada indivíduo.

```{r}
Mset <- preprocessRaw(rgSet)
```

```{r}
plotQC(getQC(Mset))
```

### Removendo as amostras com qualidade baixa

```{r}
keep <- colMeans(detP) < 0.05
```

<!-- Manter no rgset -->

```{r}
rgSet <- rgSet[,keep]
```

<!-- Manter no detP -->

```{r}
detP <- detP[,keep]
```

## Normalização

```{r, eval=FALSE}
mSetSq <- preprocessQuantile(rgSet)
```

```{r, eval=FALSE}
save(mSetSq,file="../../data/mSetSq.rds")
```

```{r}
load("../../data/mSetSq.rds")
```

```{r}
dim(mSetSq)
```

```{r,fig.width=15}
par(mfrow=c(2,1))
densityPlot(rgSet,main="Dados Brutos", legend=FALSE)
densityPlot(getBeta(mSetSq),main="Dados Pré-processados", legend=FALSE)
```

```{r, fig.width=15}
par(mfrow=c(1,2))
boxplot(getBeta(rgSet),main="Dados Brutos", legend=FALSE)
boxplot(getBeta(mSetSq),main="Dados Pré-processados", legend=FALSE)
```

```{r}
png(file = "../../../fileName3.png", width = 800, height = 600)
densityBeanPlot(getBeta(mSetSq))
dev.off()
```

```{r}
mdsPlot(getBeta(mSetSq))
```

```{r}
controlStripPlot(rgSet)
```

## Exploração de dados

```{r}
png(file = "../../../fileName.png") # defaults to 7 x 7 inches
plotMDS(getM(mSetSq), top=1000, gene.selection="common")
dev.off()
```
```{r}
png(file = "../../../fileName2.png") # defaults to 7 x 7 inches
plotMDS(getBeta(mSetSq), top=1000, gene.selection="common", main="teste")
ylab("")
dev.off()
```

## Filtragem

<!-- Filtragem das sondas problemáticas -->

```{r}
detP <- detP[match(featureNames(mSetSq),rownames(detP)),]
```

```{r}
keep <- rowSums(detP < 0.01) == ncol(mSetSq)
table(keep)
```

```{r}
mSetSqFlt <- mSetSq[keep,]
```

```{r}
remove(mSetSq)
```

<!-- Filtragem das probes dos cromossomos de sexo -->

```{r}
keep <- !(featureNames(mSetSqFlt) %in% annEPIC$Name[annEPIC$chr %in%
c("chrX","chrY")])
table(keep)
```

```{r}
mSetSqFlt <- mSetSqFlt[keep,]
```

<!-- Remove as sondas com SNP nos campos de CpG -->

```{r}
mSetSqFlt <- dropLociWithSnps(mSetSqFlt)
```

```{r}
xReactiveProbes <- xreactive_probes(array_type = "EPIC")

keep <- !(featureNames(mSetSqFlt) %in% xReactiveProbes)
table(keep)
```

```{r}
mSetSqFlt <- mSetSqFlt[keep,]
```
-->

<!-- Salvando os dados finais após todos os filtros -->

```{r}
save(mSetSqFlt,file="../../data/mSetSqFlt.rds")
```

```{r}
load("../../data/mSetSqFlt.rds")
```

<!-- Cálculo de M e Beta -->

```{r}
mVals <- getM(mSetSqFlt)
```

```{r}
bVals <- getBeta(mSetSqFlt)
```

```{r}
head(mVals[,1:5])
head(bVals[,1:5])
```
